ext/fetch/lib.rs

```diff
@@ -474,13 +476,26 @@ impl Resource for FetchRequestBodyResource {
       let nwritten = bytes.len();
       let body = RcRef::map(&self, |r| &r.body).borrow_mut().await;
       let cancel = RcRef::map(self, |r| &r.cancel);
-      body.send(Ok(bytes)).or_cancel(cancel).await?.map_err(|_| {
-        type_error("request body receiver not connected (request closed)")
-      })?;
+      body
+        .send(Some(bytes))
+        .or_cancel(cancel)
+        .await?
+        .map_err(|_| {
+          type_error("request body receiver not connected (request closed)")
+        })?;
       Ok(WriteOutcome::Full { nwritten })
     })
   }
 
+  fn shutdown(self: Rc<Self>) -> AsyncResult<()> {
+    Box::pin(async move {
+      let body = RcRef::map(&self, |r| &r.body).borrow_mut().await;
+      let cancel = RcRef::map(self, |r| &r.cancel);
+      body.send(None).or_cancel(cancel).await?.ok(); // we don't care if the receiver is closed, because this is the shutdown
```

Comment:
Not always - there is one case where hyper knows the size of the response body up front (through `content-length` header on the resp), where it will drop the body once that content length has been reached, regardless of if the stream is complete or not. This is expected behaviour, but it means that if you stream a body with an up front known size (eg a `Blob`), `shutdown` will never succeed because the body (and by extension the receiver) will have dropped by the time we try to shutdown.

I could try to wire it up to only swallow the error if this specific case has happened (might be a bit tricky to detect though).

Reply:
That's not correct. You can see this in action using this playground: https://dash.deno.com/playground/slow-sheep-54
TL;DR: Content-Length too small results in an errored response stream.

```
🦕 curl https://slow-sheep-54.deno.dev/     # No Content-Length
Hello World%                                                                    
🦕 curl https://slow-sheep-54.deno.dev/11   # Content-Length accurate.
Hello World%                                                                    
🦕 curl https://slow-sheep-54.deno.dev/100  # Content-Length larger than response.
curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
🦕 curl https://slow-sheep-54.deno.dev/1    # Content-Length smaller than response.
curl: (92) HTTP/2 stream 0 was not closed cleanly: PROTOCOL_ERROR (err 1)
```